src/openforms/formio/dynamic_config/registry.py

```diff
@@ -0,0 +1,40 @@
+from typing import Optional
+
+from openforms.plugins.plugin import AbstractBasePlugin
+from openforms.plugins.registry import BaseRegistry
+from openforms.typing import DataMapping
+
+from ..typing import Component
+
+
+class BasePlugin(AbstractBasePlugin):
+    is_enabled: bool = True
+
+    @staticmethod
+    def mutate(component: Component, data: DataMapping) -> None:  # pragma: nocover
+        ...
```

Comment:
If this is valid I learned something today. Same as `pass` I guess.

Reply:
yeah it is, often used for pure type-hinting code where you overload a function like this: 

```py
from typing import overload

@overload
def foo(arg: str) -> str: ...

@overload
def foo(arg: int) -> int: ...

def foo(arg: int | str) -> int | str:
    # actual implementation
```

it indicates that string input results in string output, int input results in int output, while without the overloads you could theoretically get one type as input and return the other as output.